0xPolygon/polygon-edge
# [Feature] Datadog profiler

This PR adds Datadog profiler into the codebase, so that it can be easily enabled on demand.
Profiler enables users to see profiles and tracing stats on their Datadog portal, but it requires some configuration of Datadog agent on the node. 

APM ( tracing ) should be enabled in Datadog agent configuration yaml file:
```
apm_config:

  enabled: true
  env: polygon-edge # this sets the environment name that will be shown on DD portal
```

To enable the profiler, a user must simply set `DD_PROFILING_ENABLED="True"` environment variable.
To set the IP and port that the DataDog agent, use `DD_AGENT_HOST` and `DD_TRACE_AGENT_PORT` environment variables.    
By default, it will try communicate with the agent on `localhost:8126`.

These env vars are now consistent with the DD doc https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/go/?tab=containers#configure-the-datadog-agent-for-apm

Additional parameters can be set by using env vars stated in the official Datadog doc https://docs.datadoghq.com/profiler/enabling/go/

# Changes include


# Checklist


## Testing


### Manual tests

Manual tests would require that the tester has access to Datadog portal.
Just set the `DD_ENABLE` env var, configure Datadog agent on the same node, and check Datadog portal if you can see your profile.

# Documentation update

Public doc will soon with the explanation on how to set this up will follow

# Additional comments

Fixes EDGE-764
